pkg/api/handlers/compat/images_push.go

```diff
@@ -187,22 +180,15 @@ loop: // break out of for/select infinite loop
 				break loop
 			}
 
-			digestBytes, err := io.ReadAll(digestFile)
-			if err != nil {
-				report.Error = &jsonmessage.JSONError{
-					Message: err.Error(),
-				}
-				report.ErrorMessage = err.Error()
-				if err := enc.Encode(report); err != nil {
-					logrus.Warnf("Failed to json encode error %q", err.Error())
-				}
-				flush()
-				break loop
-			}
 			tag := query.Tag
 			if tag == "" {
 				tag = "latest"
 			}
+
+			var digestBytes []byte
+			if pushReport != nil {
+				digestBytes = pushReport.Manifest
+			}
```

Comment:
This seems to change the API output, now you return the full manifest.
Before only the digest was printed as string. Please do not change the output of an existing field, this could break users.

Reply:
That's an oversight on my end, thanks for catching.  Yes, we need to compute the digest here.